### PR TITLE
Feature/EUI 8386

### PR DESCRIPTION
**Pull request by DavidJayakumar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-8386

### Change description ###

Query response summary page

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.ts
Comment (MateiHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.ts:1]: We aren't going to use the write tab from now on to raise a query or respond to a query (see EUI-8412).

The raise a query the user will have to go to the case page -> select the 'Raise a query' event from the dropdown in the right top corner.
To get to the respond to a query page, please append a mock qid at the end of the url.

The right component should be QueryManagementContainerComponent which is in src/cases/containers/query-management-container/query-management-container.component.ts

**Review by MateiHMCTS — CHANGES_REQUESTED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.ts
Comment (DavidJayakumar) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.ts:1]: My understanding is write tab will go but, On choosing dropdown the write-query-management-filed will be called and then raise query will be invoked if the queryItem is empty.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+@Component({
+  selector: 'ccd-query-check-your-answers',
+  templateUrl: './query-check-your-answers.component.html',
+  styleUrls: ['./query-check-your-answers.component.scss'],
+  encapsulation: ViewEncapsulation.None
+
+})
+export class QueryCheckYourAnswersComponent implements OnInit {
+
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.ts:14]: @DavidJayakumar - can remove extra blank line here (the linter will complain about this)

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.spec.ts [hunk 2]
```diff
@@ -50,5 +50,15 @@ describe('WriteQueryManagementFieldComponent', () => {
...
     expect(component).toBeTruthy();
   });
+
+  it('showResponseForm - should not show summary', () => {
+    component.showResponseForm()
+    expect(component.showSummary).toBeFalsy();
+  });
+
+  it('confirmDetails - should show summary', () => {
+    component.confirmDetails()
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.spec.ts]: @DavidJayakumar - missing semicolon

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@DavidJayakumar - some changes needed

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
+<div class="govuk-grid-row">
```
[the remark this answers is not in the document]
Comment (DavidJayakumar) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.html:1]: done

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.collection-field-table {
```
[the remark this answers is not in the document]
Comment (DavidJayakumar) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.scss:1]: done

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.ts (hunk 1 above)
Comment (DavidJayakumar) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-check-your-answers/query-check-your-answers.component.ts:14]: done

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.spec.ts (hunk 2 above)
Comment (DavidJayakumar) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.spec.ts]: This code block is moved to webapp QueryManagmentContainerComponent

**Review by DavidJayakumar — COMMENTED** [review]
@Daniel-Lam Added my comments. Can you review again?

**Review by Daniel-Lam — DISMISSED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.html [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
+      </ccd-query-write-raise-query>
+    </ng-template>
   </ng-container>
-  <ng-template #raiseQuery>
-    <ccd-query-write-raise-query
-      [formGroup]="formGroup">
-    </ccd-query-write-raise-query>
-  </ng-template>
-</div>
+  <ccd-query-check-your-answers *ngIf="showSummary" [queryItem]="queryItem" (backClicked)="showResponseForm()">
```
Comment (MateiHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/write-query-management-field.component.html]: I think we need to pass in formGroup instead?

**Review by MateiHMCTS — DISMISSED** [review]

**Review by MateiHMCTS — APPROVED** [review]

**Review by Daniel-Lam — DISMISSED** [review]
@DavidJayakumar - re-approved

**Review by johnbenjamin-hmcts — APPROVED** [review]

**Review by connorpgpmcelroy — APPROVED** [review]
LGTM

**Pull request merged**